common/app/com/commercetools/sunrise/shoppingcart/LineItemBeanFactory.java

```diff
@@ -1,22 +1,28 @@
 package com.commercetools.sunrise.shoppingcart;
 
-import com.commercetools.sunrise.common.contexts.UserContext;
 import com.commercetools.sunrise.common.ctp.ProductDataConfig;
-import com.commercetools.sunrise.common.reverserouter.ProductReverseRouter;
+import com.commercetools.sunrise.common.models.ProductAttributeBean;
+import com.commercetools.sunrise.common.models.ProductAttributeBeanFactory;
 import io.sphere.sdk.carts.LineItem;
-import io.sphere.sdk.models.Base;
 
 import javax.inject.Inject;
+import java.util.List;
 
-public class LineItemBeanFactory extends Base {
+import static java.util.stream.Collectors.toList;
+
+public class LineItemBeanFactory extends MiniCartLineItemBeanFactory {
     @Inject
     private ProductDataConfig productDataConfig;
     @Inject
-    private UserContext userContext;
-    @Inject
-    private ProductReverseRouter reverseRouter;
+    private ProductAttributeBeanFactory productAttributeBeanFactory;
 
     public LineItemBean create(final LineItem lineItem) {
-        return new LineItemBean(lineItem, productDataConfig, userContext, reverseRouter);
+        final LineItemBean bean = fillBean(new LineItemBean(), lineItem);
+        final List<ProductAttributeBean> attributes = lineItem.getVariant().getAttributes().stream()
+                .filter(attr -> productDataConfig.getSelectableAttributes().contains(attr.getName()))
+                .map(attr -> productAttributeBeanFactory.create(attr))
+                .collect(toList());
+        bean.setAttributes(attributes);
+        return bean;
```

Comment:
I would put this in another fillBean to follow the same structure everywhere, otherwise here the developer has to rewrite all the attributes part. Something like:

```
public LineItemBean create(final LineItem lineItem) {
    return fillBean(new LineItemBean(), lineItem);
}

protected <T extends LineItemBean> T fillBean(final T bean, LineItem lineItem) {
    T updatedBean = super.fillBean(bean, lineItem);
    final List<ProductAttributeBean> attributes = lineItem.getVariant().getAttributes().stream()
            .filter(attr -> productDataConfig.getSelectableAttributes().contains(attr.getName()))
            .map(attr -> productAttributeBeanFactory.create(attr))
            .collect(toList());
    updatedBean.setAttributes(attributes);
    return updatedBean;
}
```